### PR TITLE
Retrieve list of forum private messages

### PR DESCRIPTION
**Pull request by gberaudo**
- split non sso code related to discourse in a separate file;
- add a POST method for retrieving list of all or unread  messages.

### CODE REVIEW

**Comment by asaunier** [comment]
Thanks @gberaudo 
Actually we don't need the detailed list of messages, only the number of unread messages (will be shown in the UI near the user name or avatar, as a "new unread messages" indicator). We also need a way to redirect to the MP inbox when the user clicks on some button.

**Comment by gberaudo** [comment]
@asaunier, updated the PR to count and return the number of unread messages.
The link to the forum is returned together so that it is straightforward to integrate in the UI.

**Comment by asaunier** [comment]
OK, thanks for the change.

File: c2corg_api/views/forum.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        client = get_discourse_client(settings)
+        d_username = discourse_get_username_by_userid(client, userid)
+        messages = client.private_messages_unread(d_username)
+
+        count = len(messages['topic_list']['topics'])
+        link = '%s/users/%s/messages' % (
+            get_discourse_public_url(settings), d_username)
+
+        return dict(
```
Comment (tsauerwein) [inline c2corg_api/views/forum.py]: Minor: elsewhere we are using `{link: link, count: count}`

**Comment by asaunier** [comment]
Should we advertise somewhere this new ``/forum/private-messages/unread-count`` service or do you think having a look at the ``view_config`` in the ``forum`` view is enough?

**Comment by tsauerwein** [comment]
LGTM

File: c2corg_api/views/forum.py (hunk 1 above)
Comment (gberaudo) [inline c2corg_api/views/forum.py]: OK, updated.

**Comment by gberaudo** [comment]
Thanks for the review @asaunier and @tsauerwein.

**Pull request merged**